backend/src/Controller/Admin/SupplierCategory/AdminSupplierCategoryController.php

```diff
@@ -0,0 +1,254 @@
+            return $this->response($response, self::SUPPLIER_CATEGORY_NOT_EXIST);
+        }
+
+        return $this->response($response, self::UPDATE);
+    }
+
+    /**
+     * Admin: update existing supplier category status.
+     * @Route("suppliercategorystatus", name="updateSupplierCategoryStatusByAdmin", methods={"PUT"})
+     * @IsGranted("ROLE_ADMIN")
+     * @param Request $request
+     * @return JsonResponse
+     *
+     * @OA\Tag(name="Supplier Category")
+     *
+     * @OA\Parameter(
+     *      name="token",
+     *      in="header",
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\RequestBody(
+     *      description="Create new supplier category by admin",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="integer", property="id"),
+     *          @OA\Property(type="boolean", property="status")
+     *      )
+     * )
+     *
+     * @OA\Response(
+     *      response=201,
+     *      description="Returns the updated supplier category",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="object", property="Data",
+     *                  @OA\Property(type="integer", property="id"),
+     *                  @OA\Property(type="string", property="name"),
+     *                  @OA\Property(type="string", property="description"),
+     *                  @OA\Property(type="boolean", property="status")
+     *          )
+     *      )
+     * )
+     *
+     * or
+     *
+     * @OA\Response(
+     *      response="default",
+     *      description="Returns the updated supplier category",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code", example="9550"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="object", property="Data", example="supplierCategoryNotExist"
+     *          )
+     *      )
+     * )
+     *
+     * @Security(name="Bearer")
+     */
+    public function updateSupplierCategoryStatusByAdmin(Request $request): JsonResponse
+    {
+        $data = json_decode($request->getContent(), true);
+
+        $request = $this->autoMapping->map(stdClass::class, SupplierCategoryStatusUpdateByAdminRequest::class, (object)$data);
+
+        $violations = $this->validator->validate($request);
+        if (\count($violations) > 0) {
+            $violationsString = (string) $violations;
+
+            return new JsonResponse($violationsString, Response::HTTP_OK);
+        }
+
+        $response = $this->adminSupplierCategoryService->updateSupplierCategoryStatusByAdmin($request);
+
+        if ($response === SupplierCategoryResultConstant::SUPPLIER_CATEGORY_NOT_EXIST) {
+            return $this->response($response, self::SUPPLIER_CATEGORY_NOT_EXIST);
```

Comment:
please, don't return '$response' when error ,but return 'MainErrorConstant::ERROR_MSG'  because no display 'Data' in response